### PR TITLE
Send & receive SCP quorum sets to fix SCP message propagation

### LINKED ISSUE #606: SCP messages need to be broadcasted to all a Node's connected Validator peers
In the current design a Validator is broadcasting messages to its connected peers through this function:

https://github.com/bpfkorea/agora/blob/acbd5464cd735d6c62910d49e2bac7346926ceb0/source/agora/consensus/protocol/Nominator.d#L334-L338

However these list of peers are a subset of Validators. They are the quorum slice of the node itself.

Essentially, given node A which has nodes B and C in its quorum slice:

```
A -> {B, C}
```

Node `A` will broadcast SCP messages (Nominate, etc..) to nodes `B` and `C`. In fact in this case, nodes `B` and `C` don't care about messages from node `A`, because `A` is not in either of their quorum slices.

Additionally, if there is any node `X` which has node `A` in its quorum set but node `A` does not have node `X` in its quorum set, then node `X` will not receive nomination messages from node `A`.

Example:

```
X -> A -> {B, C}
```

Node `X` must receive SCP messages from node `A`. In the current design, it does not.

In the test-suite all the nodes have each other in their quorum slice so this problem was not immediately apparent.

Reading https://www.stellar.org/developers-blog/intuitive-stellar-consensus-protocol is what made me realize there was a problem.

-----

What needs to be changed:

- [ ] The list of peers in the `Nominator` should be the list of Validator nodes which connected to that node, expecting SCP nomination / balloting statements.
- [ ] The connection may be established at any arbitrary time. Despite being possible to determine the exact quorum configuration based on https://github.com/bpfkorea/agora/issues/240, a Validator node A can only know the public key of such node, it cannot know for sure how node X with some public key _XP_ will connect to it (which IP, etc).

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Fixes https://github.com/bpfkorea/agora/issues/606

Requires: 
- ~https://github.com/bpfkorea/agora/pull/643~
- ~https://github.com/bpfkorea/agora/pull/648~

### CODE REVIEW

**Comment by AndrejMitrovic** [comment]
Some of these commits should be split off into another PR.

File: source/agora/network/NetworkManager.d [hunk 1]
```diff
@@ -365,6 +381,8 @@ public class NetworkManager
             try
             {
                 node.handshake(this.getAddress());
+                if (node.quorum_hash != StellarHash.init)
+                    this.quorum_sets[node.quorum_hash] = node.quorum_set;
```
Comment (AndrejMitrovic) [inline source/agora/network/NetworkManager.d:396]: Actually this should be lazy. If the quorum hash is already known, no need to retrieve the quorum set.

**Comment by AndrejMitrovic** [comment]
I will split this into two, and remove the necessity to exchange SCP structs in the API.

**Comment by AndrejMitrovic** [comment]
This needs to be reworked with regards to https://github.com/bpfkorea/agora/issues/702.

File: source/agora/node/Node.d [hunk 2]
```diff
@@ -270,6 +270,55 @@ public class Node : API
...
+    ***************************************************************************/
+
+    public override SCPQuorumSet getQuorumSet (Hash hash) @trusted
+    {
+        import scpd.types.Stellar_types;
+        if (this.config.node.is_validator)
+        {
+            auto stellar_hash = uint512(hash);
+            auto set = this.nominator.getQSet(stellar_hash);
+            assert(set.ptr !is null);
```
Comment (TrustHenry) [inline source/agora/node/Node.d:313]: If StellarHash is not new, an error occurs.
Can it be used for malicious attacks?

**Comment by AndrejMitrovic** [comment]
As mentioned in https://github.com/bpfkorea/agora/pull/621#issuecomment-609649242, this PR has become unnecessary. We don't need to look up quorum sets explicitly because they will be known in advance once #240 is implemented.

**Pull request closed without merging**